### DISCUSSION TITLE
`azure.yaml` schema change to support other languages (e.g. ruby) or to allow devs to build support for other languages

### DISCUSSION THREAD

**Discussion by kurt-mueller-osumc** [Azure/azure-dev#1637] [Ideas]
I have a Ruby on Rails project that I've been provisioning infrastructure (App Service, Container Registry) for via Azure Bicep.

I'd like to be able to provision and deploy this application via `azd`. To that end, even if there are no official plans to build language support for ruby, as an end-user, I'd like the ability to have a way to configure this myself.

Are there any plans to allow end-users to specify their own custom language in `azure.yaml`?

**Comment by craigofnz** [comment]
From other cases where language is not supported I've been able to use postprovision hooks. For example, for IT operational tasks using Powershell based functions.

However, this does not use azd idioms like `azd deploy`, and a pre or post deploy hook cannot be used for a service generates an error due to using an unsupported language (or other issues for otherwise deployable resources or artifacts that are not supported by azd at this time).
